### ISSUE TITLE
🐛 BUG: pages dev ignoring function when using a a command

### ISSUE THREAD

**Issue by PierBover** [cloudflare/workers-sdk#2964]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version of `Wrangler` are you using?

2.13.0

### What operating system are you using?

Mac

### Describe the Bug

When using `npx wrangler pages dev -- <COMMAND>` both the `/functions` directory and the `_worker.js` are being ignored.

When opening the `http://127.0.0.1:8788/_worker.js` the file is indeed being served as plain text.

<img width="515" alt="image" src="https://user-images.githubusercontent.com/226483/228912177-b76235ce-2197-4483-b139-9ec61f9ecfa2.png">

**Comment by jahands** [comment]
@PierBover Thanks for the report!
Unfortunately, the `-- <COMMAND>` syntax does not currently support metadata files in the output directory (eg. `_worker.js`, `_headers`, etc). You can work around the `_worker.js` issue by manually specifying the path: `--script-path ./public/_worker.js`, but that may not be a great solution if you have other Pages metadata files in your output directory.

An alternative approach would be to not use the proxy and do something like what Remix does:
```json
{
  "scripts": {
    "dev:remix": "cross-env NODE_ENV=development remix watch",
    "dev:wrangler": "cross-env NODE_ENV=development wrangler pages dev ./public",
    "dev:css": "tailwindcss -w -i ./styles/app.css -o app/styles/app.css",
    "dev": "cross-env NODE_ENV=development remix build && run-p dev:*"
  }
}
```
This will also prevent any unexpected behavior where the framework dev server returns assets differently from what Pages does.

**Comment by PierBover** [comment]
Thanks @jahands using multiple server in parallel would probably work for some use cases but it's not a solution.

For instance, how do you submit a `<form>` from the SSG like Astro to a Cloudflare function?

Having two servers running in parallel means the hosts/ports would be different compared to production where everything would run from the same domain. So now we need a special setup to handle dev vs production. Also, what about CORS, SSL, cookies, etc?

During dev all this should really be transparent for the browser and running from a single host/port to avoid all those issues.

**Comment by lrapoport-cf** [comment]
hi @PierBover :) we haven't heard from you in a while, so we’re going to close this issue for now. if the solutions noted above by @jahands amd @CarmenPopoviciu didn't work and this is still a problem, please feel free to open a new issue referencing this one. please be sure to include an updated summary and any additional requested details, followup, or context and we will revisit and investigate further. thanks!
